dot/parachain/collator-protocol/validator_side.go

```diff
@@ -440,21 +457,106 @@ func (cpvs CollatorProtocolValidatorSide) processMessage(msg interface{}) error
 		// TODO: handle network message https://github.com/ChainSafe/gossamer/issues/3515
 		// https://github.com/paritytech/polkadot-sdk/blob/db3fd687262c68b115ab6724dfaa6a71d4a48a59/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L1457 //nolint
 	case SecondedOverseerMsg:
-		// TODO: handle seconded message https://github.com/ChainSafe/gossamer/issues/3516
-		// https://github.com/paritytech/polkadot-sdk/blob/db3fd687262c68b115ab6724dfaa6a71d4a48a59/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L1466 //nolint
+		statementV, err := msg.Stmt.Payload.Value()
+		if err != nil {
+			return fmt.Errorf("getting value of statement: %w", err)
+		}
+		if statementV.Index() != 1 {
+			return fmt.Errorf("expected a seconded statement")
```

Comment:
Given the rust reference if the statement payload is not a `Seconded` message it doesn't return an error but just log some info and returns without error, so why do we need to return an error here? The caller is expecting this error?

ref: https://github.com/paritytech/polkadot-sdk/blob/f4827dcd16fc5585a8d15c6a32bc27ecde00e7f9/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L1469
```rust
gum::warn!(
  target: LOG_TARGET,
  ?stmt,
  relay_parent = %parent,
  "Seconded message received with a `Valid` statement",
);
```

Reply:
>  it doesn't return an error but just log some info and returns without error,

That's a common pattern I have seen throughout polkadot-sdk project.  We don't necessarily have to follow that pattern. I have chosen to follow go's common pattern of returning the error.

All the errors from different functions used here would end up in `Run` function. In `Run` function we log it without returning(that way one error in one task doesn't stop the entire subsystem). Logging at in error seems a bit more cleaner to me, and it would have added context of error propagation.